test/test_utils.py

```diff
@@ -960,11 +783,166 @@ def test_build_container_default_gcc(tmpdir):
         use_host_conda_bld=True,
         image_build_dir=image_build_dir,
     )
+
+    pkg_paths = utils.built_package_paths(r.recipe_dirs['one'])
     build_result = build.build(
         recipe=r.recipe_dirs['one'],
         recipe_folder='.',
-        env={},
+        pkg_paths=pkg_paths,
         docker_builder=docker_builder,
         mulled_test=False,
     )
     assert build_result.success
+
+
+def test_conda_forge_pins(caplog):
+    caplog.set_level(logging.DEBUG)
+    r = Recipes(
+        """
+        one:
+          meta.yaml: |
+            package:
+              name: one
+              version: 0.1
+            requirements:
+              run:
+                - zlib {{ zlib }}
```

Comment:
We should aiming for not including zlib at all at run-time. This can be archived with run_exports, see here for example: https://github.com/bioconda/bioconda-recipes/pull/8831

However, this means we need to wait a little bit longer -  conda-forge is working on this atm.

Reply:
Thanks Bjoern. I approached this entirely from reading the conda docs. It wasn't clear to me from the docs that's what `run_exports` is doing -- that's really nice!